am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java

```diff
@@ -0,0 +1,98 @@
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toSet;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.amlib.helpers.DefaultRoleSetupDataFactory.grantMultipleResources;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.CREATE_PERMISSION;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.READ_PERMISSION;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_TYPE;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ROLE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ROLE_NAMES;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.SERVICE_NAME;
+
+class GetRolePermissionsIntegrationTest extends IntegrationBaseTest {
+    private static AccessManagementService ams;
+    private static DefaultRoleSetupImportService defaultRoleService;
+
+    @BeforeAll
+    static void setUp() {
+        ams = new AccessManagementService(db.getJdbcUrl(), db.getUsername(), db.getPassword());
+        defaultRoleService = new DefaultRoleSetupImportService(db.getJdbcUrl(), db.getUsername(), db.getPassword());
+    }
+
+    @BeforeEach
+    void roleSetUp() {
+
+        defaultRoleService.addRole(ROLE_NAME, RoleType.RESOURCE, SecurityClassification.PUBLIC, AccessType.ROLE_BASED);
+        defaultRoleService.addService(SERVICE_NAME);
+        defaultRoleService.addResourceDefinition(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME);
+        defaultRoleService.grantDefaultPermission(DefaultPermissionGrant.builder()
+            .roleName(ROLE_NAME)
+            .serviceName(SERVICE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .resourceName(RESOURCE_NAME)
+            .attributePermissions(grantMultipleResources())
+            .build());
+
+
+    }
+
+    @Test
+    void returnListOfPermissionsForRoleName() {
+        Map<JsonPointer, Set<Permission>> accessRecord = ams.getRolePermissions(SERVICE_NAME,
+            RESOURCE_TYPE, RESOURCE_NAME, ROLE_NAMES);
+
+        assertThat(accessRecord).containsEntry(JsonPointer.valueOf("/test"), READ_PERMISSION);
+        assertThat(accessRecord).containsEntry(JsonPointer.valueOf("/test2"), READ_PERMISSION);
+        assertThat(accessRecord).containsEntry(JsonPointer.valueOf("/testCreate"), CREATE_PERMISSION);
+
+
+        assertThat(accessRecord).hasSize(3);
+    }
+
+    @Test
+    void shouldReturnNullWhenServiceNameDoesNotExist() {
+        Map<JsonPointer, Set<Permission>> accessRecord = ams.getRolePermissions("Service 2",
+            RESOURCE_TYPE, RESOURCE_NAME, ROLE_NAMES);
+        assertThat(accessRecord).isNull();
+    }
+
+    @Test
+    void shouldReturnNullWhenResourceTypeDoesNotExist() {
+        Map<JsonPointer, Set<Permission>> accessRecord = ams.getRolePermissions(SERVICE_NAME,
+            "No Resource Type ", RESOURCE_NAME, ROLE_NAMES);
```

Comment:
Please use "Unknown" instead of "No" e.g. "Unknown resource type". It is more expressive IMO. Same in other tests.

Reply:
👍